.github/workflows/pytest.yml

```diff
@@ -86,44 +99,43 @@ jobs:
       fail-fast: false
       matrix:
         os: [macos-latest, windows-latest]
-        py: ['3.8']
+        py: ['3.9']
     steps:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0 # Fetch all history for all tags and branches
-      - uses: actions/setup-python@v3
-        with:
-          python-version: ${{ matrix.py }}
 
-      - uses: actions/cache@v3
-        if: startsWith(runner.os, 'macOS')
+      - uses: CagtayFabry/pydeps2env@main
         with:
-          path: ~/Library/Caches/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('setup.cfg') }}
-          restore-keys: |
-            ${{ runner.os }}-pip-
+          channels: 'conda-forge defaults'
+          extras: 'test'
+          setup_requires: 'include'
 
-      - uses: actions/cache@v3
-        if: startsWith(runner.os, 'Windows')
+      - uses: mamba-org/provision-with-micromamba@main
         with:
-          path: ~\AppData\Local\pip\Cache
-          key: ${{ runner.os }}-pip-${{ hashFiles('setup.cfg') }}
-          restore-keys: |
-            ${{ runner.os }}-pip-
+          environment-file: ./environment.yml
+          environment-name: weldx
+          cache-env: true
+          extra-specs: |
+            python=${{ matrix.py }}
+            wheel
+            pip
+
+      - name: activate env
+        run: micromamba activate weldx
 
       - name: pip installs
         run: |
-          pip install wheel
-          pip install -e .[test]
+          python -m pip install -e .
 
       - name: setup matplotlib
         if: startsWith(runner.os, 'Windows')
         run: |
-          if not exist %userprofile%\.matplotlib\ ( mkdir %userprofile%\.matplotlib\ )
-          echo backend: Agg > %userprofile%\.matplotlib\matplotlibrc
+          mkdir $env:USERPROFILE\.matplotlib\
+          echo backend: Agg > $env:USERPROFILE\.matplotlib\matplotlibrc
           python -c "import matplotlib as m; print(m.matplotlib_fname())"
           python -c "import matplotlib as m; print(m.rc_params()['backend'])"
-        shell: cmd
+        shell: pwsh
```

Comment:
unfortunately I have no idea how to get this running with `shell: bash`

the `cmd` doesn't know about the conda environment so it doesn't work...

Reply:
I have removed the debug code lines so no python/conda is required, it works now